dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -239,6 +254,53 @@ def test_correct_duplicate_row_count_test(self):
         self.assertEqual(2999, self.trained_schema.total_samples)
         self.assertEqual(0.0, self.trained_schema._get_duplicate_row_count())
 
+    def test_correlation(self):
+        # data with one column, correlation not updated
+        data = pd.DataFrame([1.0, None, 1.0, None, 5.0])
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.correlation_matrix)
+
+        # data with two columns, but only one is numerical,
+        # then correlation is not updated
+        data = pd.DataFrame([
+            ['test1', 1.0],
+            ['test2', None],
+            ['test1', 1.0],
+            [None, None]])
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.correlation_matrix)
+
+        # data with multiple numerical columns
+        data = pd.DataFrame({'a': [3, 2, 1, 7, 5, 9, 4, 10, 7, 2],
+                             'b': [10, 11, 1, 4, 2, 5, 6, 3, 9, 8],
+                             'c': [1, 5, 3, 5, 7, 2, 6, 8, 1, 2]})
+        profiler = dp.StructuredProfiler(data)
+        expected_corr_mat = data.corr()
+        pd.util.testing.assert_frame_equal(expected_corr_mat,
+                                           profiler.correlation_matrix)
+
+        # data with multiple numerical columns, with nan values
+        data = pd.DataFrame({'a': [np.nan, np.nan, 1, 7, 5, 9, 4, 10, 7, 2],
+                             'b': [10, 11, np.nan, 4, 2, 5, 6, 3, 9, 8],
+                             'c': [1, 5, 3, 5, 7, 2, 6, 8, np.nan, np.nan]})
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.correlation_matrix)
+
+    def test_merge_correlation(self):
+        data = pd.DataFrame({'a': [3, 2, 1, 7, 5, 9, 4, 10, 7, 2],
+                             'b': [10, 11, 1, 4, 2, 5, 6, 3, 9, 8],
+                             'c': [1, 5, 3, 5, 7, 2, 6, 8, 1, 2]})
+        data1 = data[:int(len(data) / 2)]
+        data2 = data[int(len(data) / 2):]
+
+        profile1 = dp.StructuredProfiler(data1)
+        profile2 = dp.StructuredProfiler(data2)
+        merged_profile = profile1 + profile2
+
+        expected_corr_mat = data.corr()
+        pd.util.testing.assert_frame_equal(expected_corr_mat,
+                                           merged_profile.correlation_matrix)
+
```

Comment:
I think this needs further discussion. I think it would be a silent confusing if the user receives a report that when merged only reflects one of the profiles. 

IMO, the correlations should be set to `None` in this case as we cannot compute the merge and the new correlations might actually be drastically different that the chosen values.

@ChrisWallace2020 @granteden @lettergram thoughts?

Reply:
I don't really follow your comment. Why the correlation should be set to None in this case. Two halves of the data both have correlations, then we merge from them?

Reply:
for merge makes sense.

Reply:
added the test for stream